test/unit/full_text_search/change_git_test.rb

```diff
@@ -79,9 +68,8 @@ def test_fetch_changesets
                      },
                    ],
                    [
-                     records.collect(&:title),
```

Comment:
The `new_file.txt` file is removed on Git repository's test data in master.
So we remove this assertions because I think it's enough to test search function by checking first record and last record.

Reply:
件数があっているか確認したいので残してください。

Reply:
ファイルの順序自体は変わってしまうのですが、件数は取れるのでその方針で残します。
- 消してしまうと検索結果の件数が変わらないことを把握できなくなってしまうため

Reply:
Redmine master だけ新しく更新されたファイルがあり、それを検索対象から除いているので件数が変わる。
少し対応作を考えます。

Reply:
どうやろうとしています？
`target_records.collect(&:title)`にするだけでいい気がするんですけど。

Reply:
新規で更新されたふぁいるである「"new_file.txt"」の影響でテスト結果がずれてしまうので、
それだけを除いた検索に変更すれば全て丸く収まりそう。

Reply:
fix: [7d89019](https://github.com/clear-code/redmine_full_text_search/pull/157/commits/7d89019175519e7f3f2316ae865014349fa18278) 新規テストデータで更新され影響のあったファイルを取り除くことでテストの整合性を維持する形で修正しました。

Reply:
ここの検索対象ってリポジトリーの各コミットですよね？
新しいコミットが増えたからといって過去のコミットが消えるとは思えないんですけど。。。
Redmineのmasterのテスト用のGitリポジトリーは新しいコミットが増えているだけじゃなくて既存のコミットも変更しているんですか？

Reply:
下記のようになります。
```ruby
target_records.collect(&:title) # last_modified_at: ..Time.parse("2024-07-01T00:00:00Z")
[
  "images/edit.png",
  "copied_README",
  "renamed_test.txt",
  "sources/watchers_controller.rb",
  "this_is_a_really_long_and_verbose_directory_name/this_is_because_of_a_simple_reason/it_is_testing_the_ability_of_redmine_to_use_really_long_path_names/These_names_exceed_255_chars_in_total/That_is_the_single_reason_why_we_have_this_directory_here/But_there_might_also_be_additonal_reasons/And_then_there_is_not_even_somthing_funny_in_here.txt",
  "filemane with spaces.txt",
  " filename with a leading space.txt ",
  "latin-1/test00.txt",
  "README",
  "latin-1-dir/make-latin-1-file.rb",
  "issue-8857/test00.txt",
  "issue-8857/test01.txt"
]
```

Reply:
```ruby
target_records.collect(&:title) # without last_modified_at: ..Time.parse("2024-07-01T00:00:00Z")
[
  "images/edit.png",
  "copied_README",
  "renamed_test.txt",
  "sources/watchers_controller.rb",
  "this_is_a_really_long_and_verbose_directory_name/this_is_because_of_a_simple_reason/it_is_testing_the_ability_of_redmine_to_use_really_long_path_names/These_names_exceed_255_chars_in_total/That_is_the_single_reason_why_we_have_this_directory_here/But_there_might_also_be_additonal_reasons/And_then_there_is_not_even_somthing_funny_in_here.txt",
  "filemane with spaces.txt",
  " filename with a leading space.txt ",
  "latin-1/test00.txt",
  "README",
  "latin-1-dir/make-latin-1-file.rb",
  "issue-8857/test00.txt",
  "issue-8857/test01.txt",
  "new_file.txt"
]
```

Reply:
なんでそうなるの！？

Reply:
調べ方はわかります？

Reply:
```ruby
> pp changes
[#<Change:0x00005a4472638ab0
  id: 968,
  changeset_id: 658,
  action: "A",
  path: "new_file.txt",
  from_path: nil,
  from_revision: nil,
  revision: nil,
  branch: nil>,
 #<Change:0x00005a44726389e8
  id: 990,
  changeset_id: 675,
  action: "M",
  path: "new_file.txt",
  from_path: nil,
  from_revision: nil,
  revision: nil,
  branch: nil>]

> pp Target.where(container_id: repository.id,
                              container_type_id: Type.repository.id).
                   order(source_id: :asc).
                   collect(&:title)
["images/edit.png",
 "copied_README",
 "renamed_test.txt",
 "sources/watchers_controller.rb",
 "this_is_a_really_long_and_verbose_directory_name/this_is_because_of_a_simple_reason/it_is_testing_the_ability_of_redmine_to_use_really_long_path_names/These_names_exceed_255_chars_in_total/That_is_the_single_reason_why_we_have_this_directory_here/But_there_might_also_be_additonal_reasons/And_then_there_is_not_even_somthing_funny_in_here.txt",
 "filemane with spaces.txt",
 " filename with a leading space.txt ",
 "latin-1/test00.txt",
 "README",
 "latin-1-dir/make-latin-1-file.rb",
 "issue-8857/test00.txt",
 "issue-8857/test01.txt",
 "new_file.txt"]
```

Reply:
各ファイルの最新の状態に結びついた Target だけが存在しているように見える。
期待値としては、変更ごとに Target がいてほしいはず。
```console
pp Target.where(title: "new_file.txt")
[#<FullTextSearch::Target:0x00005a4472e2e710
  id: 1100,
  source_id: 990,
  source_type_id: 89,
  project_id: 3,
  container_id: 13,
  container_type_id: 90,
  custom_field_id: nil,
  is_private: nil,
  last_modified_at: Mon, 08 Jul 2024 17:15:16.000000000 UTC +00:00,
  title: "new_file.txt",
  content: "This is a brand new file!\n",
  tag_ids: [239],
  registered_at: Mon, 08 Jul 2024 17:15:16.000000000 UTC +00:00>]
```

Reply:
あぁ、そういうことかー
じゃあ、`last_modified_at`で絞り込んでも`new_file.txt`を除外してもstableにはならないっすね。
新しいコミットで変更が入ったらまた結果が変わる可能性がある。
違うアプローチにしないとダメだ。

Reply:
そうですね。
既存のファイルに対して変更が入るたびに更新する必要がありそうです。

補足: このファイルに変更が入る頻度はかなり低そうです。
https://github.com/redmine/redmine/commits/master/test/fixtures/repositories/git_repository.tar.gz

Reply:
fix: 7d89019...e101a8d テストデータが更新されている master の場合だけ、テストの期待値を変更するよう修正しました。
- テストデータが更新された場合でも master 側の期待値だけを修正するだけで済むため